pop_tools/zonal_average.py

```diff
@@ -0,0 +1,212 @@
+import os
+import warnings
+
+import numpy as np
+import xarray as xr
+import xesmf as xe
+
+from .grid import get_grid
+
+
+def _generate_dest_grid(dy=None, dx=None, method_gen_grid='regular_lat_lon'):
+    """
+    Generates the destination grid
+
+    Parameters
+    ----------
+    dy: float
+      Horizontal grid spacing in y-direction (latitudinal)
+
+    dy: float
+      Horizontal grid spcaing in x-direction (longitudinal)
+    """
+
+    # Use regular lat/lon with regular spacing
+    if method_gen_grid == 'regular_lat_lon':
+        if dy is None:
+            dy = 0.25
+
+        if dx is None:
+            dx = dy
+
+    # Able to add other options at a later point
+
```

Comment:
We may want to add a warning just in case the user provides `method_gen_grid` option that isn't supported. Alternatively, we could just raise an error.